plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeTableChangesWithAccessTracking.java

```diff
@@ -0,0 +1,114 @@
+import io.trino.Session;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.plugin.deltalake.metastore.TestingDeltaLakeMetastoreModule;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static com.google.inject.util.Modules.EMPTY_MODULE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDeltaLakeTableChangesWithAccessTracking
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_delta_lake_table_changes_tracking_" + randomNameSuffix();
+
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("delta_lake")
+            .setSchema(SCHEMA)
+            .build();
+
+    private File dataDirectory;
+    private HiveMetastore metastore;
+    private AccessTrackingFileSystemFactory fileSystemFactory;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake").toFile();
+        metastore = createTestingFileHiveMetastore(dataDirectory);
+        fileSystemFactory = new AccessTrackingFileSystemFactory(new HdfsFileSystemFactory(HDFS_ENVIRONMENT));
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.of(new TestingDeltaLakeMetastoreModule(metastore)), Optional.of(fileSystemFactory), EMPTY_MODULE));
+        queryRunner.createCatalog(
+                "delta_lake",
+                "delta_lake",
+                ImmutableMap.of("delta.enable-non-concurrent-writes", "true"));
+
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void cleanup()
+            throws IOException
+    {
+        if (metastore != null) {
+            metastore.dropDatabase(SCHEMA, false);
+            deleteRecursively(dataDirectory.toPath(), ALLOW_INSECURE);
+        }
+    }
+
+    @Test
+    public void testTableChangesFileSystemAccess()
+    {
+        String tableName = "test_reading_ranges_of_changes_on_table_with_cdf_enabled_" + randomNameSuffix();
```

Comment:
- sync the table name with test name 
- this is a local, isolated test. you can use fixed short table name. this makes it easier to read test queries.